include/knowhere/sparse_utils.h

```diff
@@ -274,4 +274,84 @@ class MaxMinHeap {
     std::vector<SparseIdVal<T>> pool_;
 };  // class MaxMinHeap
 
+// A std::vector like container but uses fixed size free memory(typically from
+// mmap) as backing store.
+//
+// It supports limited operations of std::vector.
+//
+// Must be initialized with a valid pointer to memory when used. The memory must be
+// valid during the lifetime of this object. After initialization, MmapVector will
+// have space for mmap_element_count_ elements, none of which are initialized.
+//
+// Currently only used in sparse InvertedIndex. Move to other places if needed.
+template <typename T>
+class MmapVector {
```

Comment:
what is the difference between `MmapVector` and https://en.cppreference.com/w/cpp/container/span ?
Also, technically, this is not a `MmapVector`, but rather a `VectorView` (similar to the relation between `std::string` and `std::string_view`)

Reply:
CMIIW `std::span` is more like referencing a list of initialized objects? so `std::span()::size()` always equals to its capacity. while in this case the contents of the containers are built up gradually using `emplace_back` and `size()` grows from 0 to capacity. though the initialization is still all done in the `Load` method. 

If we want to use `std::span`:

```c++
template <typename T>
class FixedVector : public std::span<T> {

    emplace_back() {  // to have a consistent API like vector so we can mix the usage with std::vector
    }

    size() override {
        // return number of constructed elements instead of `byte_size / sizeof(T)`.
    }
}
```

Another option is to avoid overloading the meaning of `size()`(and also avoid subclassing it) by first placement new all objects on the raw ptr, and then just create the spans out of them. But then we won't be able to reuse `add_row_to_index`.

WDYT? 

And yes it is not a `MmapedVector` indeed. But it is also not a view: it is more like an append-only vector with pre-allocated unextendable memory. How about `FixedVector`?

Reply:
I see. Basically, I got confused by the name of this class :) It totally makes sense to have this separate entity.
Naming is tricky
* it must include the word `view` / `span` to indicate that an object instance does not own the memory (unlike vector or array that do)
* the word `vector` needs to be included to indicate that this is a contiguous data chunk
* the word `array` is less appropriate, bcz it usually indicates a fixed size, not kinda `growable` as of in our use case

Maybe, something like `VectorLikeView` or `GrowableVectorView`?

Reply:
Using `GrowableVectorView`, liking it over others.